controllers/snapshot/snapshot_adapter.go

```diff
@@ -222,10 +222,20 @@ TestScenarioLoop:
 			a.logger.Error(err, "Failed to create snapshotEnvironmentbinding for snapshot.",
 				"binding", binding,
 				"snapshot.Spec.Components", a.snapshot.Spec.Components)
+			return reconciler.RequeueOnErrorOrStop(err)
 		}
 		a.logger.LogAuditEvent("A snapshotEnvironmentbinding is created", binding, h.LogActionAdd,
 			"environment.Name", copyEnv.Name,
 			"integrationTestScenario.Name", integrationTestScenario.Name)
+		err = a.labelBindingWithScenario(binding, integrationTestScenario.Name)
```

Comment:
Hm, thinking some more about this - if we fail on adding the label to the binding here and requeue the reconcile loop for this, we will find the existing environment [here](https://github.com/redhat-appstudio/integration-service/pull/218/files#diff-8e718c6bf06f754db7e9e94c6f74b5c6704941dff95d637289e3c4cc194c95a0R187) and will not get to this line again.
Should we also check if the appropriate SnapshotEnvironmentBinding exists with the correct labels below [that line](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/snapshot/snapshot_adapter.go#L187)?

Reply:
Good point. Thanks!
So I will check if the binding exists and label exists after [this line](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/snapshot/snapshot_adapter.go#L187)

Reply:
Updated. Thanks!

Reply:
I found the udpated code is not shown. Let me check again.

Reply:
It can be reviewed now.

Reply:
Thanks!